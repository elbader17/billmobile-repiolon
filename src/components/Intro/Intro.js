//Component that renders the introduction of the application
import React, { Component } from 'react';
import { Text, StyleSheet } from 'react-native';
import Swiper from 'react-native-swiper'; //https://github.com/leecade/react-native-swiper
import Screen1 from './Screens/Screen1';
import Screen2 from './Screens/Screen2';
import Screen3 from './Screens/Screen3';

class Intro extends Component {
  render() {
    return (
      //Configuration of component properties
      <Swiper
        showsButtons
        loop={false} 
        dotColor={gray} 
        style={styles.wrapper} 
        activeDotColor={white} 
        buttonWrapperStyle = {positionButtonNext}
        prevButton = {<Text style={styles.prev}></Text>} 
        nextButton = {<Text style={styles.styleButton}>Siguiente</Text>} 
      >
        <Screen1 />
        <Screen2 /> 
        <Screen3 /> 
      </Swiper>
    );
  }
}

var positionButtonNext = { top: 202, justifyContent: 'center'};
const white = '#fff';
const gray = '#C4C4C4';

const styles = StyleSheet.create({
  wrapper: {},
  styleButton: {
    width: 247, 
    height: 40, 
    padding: 10, 
    borderRadius: 2, 
    borderWidth: 1, 
    borderColor: white, 
    color: white,
    textAlign: 'center',
    fontFamily: 'Lato-Semibold', 
    fontSize: 14, 
  },
  prev: {}
});

export default Intro;