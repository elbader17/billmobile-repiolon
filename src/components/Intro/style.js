import { StyleSheet, Dimensions } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../constants/colors';
import { FONTS } from '../../constants/fonts';

const styles = StyleSheet.create({
  silde: {
    flex: 1,
    backgroundColor: 'transparent',
    alignItems: 'center',
  },
  box1: {
    flex: 0.8,
    justifyContent: 'center',
  },
  box2: {
    flex: 0.2,
    justifyContent: 'center',
  },
  box3: {
    flex: 1,
    paddingHorizontal: 50
  },
  number: {
    fontFamily: FONTS.regular,
    fontSize: FONTS.size78,
    color: COLORS.white,
  },
  textTittle: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size18,
    color: COLORS.white,
    textAlign: 'center',
    width: wp('80%'),
    lineHeight: 32,
  },
  textDescription: {
    fontFamily: FONTS.latoLight,
    fontSize: FONTS.size16,
    color: COLORS.white,
    textAlign: 'center',
    width: wp('85%'),
    lineHeight: 21,
  },
  activeDotStyle: {
    backgroundColor: COLORS.white,
    width: 9,
    height: 9
  },
  dotStyle: {
    backgroundColor: COLORS.gray,
    width: 9,
    height: 9
  },
  button: {
    backgroundColor: 'transparent',
    borderColor: COLORS.white, 
    borderRadius: 30, 
    borderWidth: 1,
  },
  buttonText: {
    fontFamily: FONTS.latoRegular, 
    fontSize: FONTS.size16, 
    color: COLORS.white,
  }
});

export default styles;