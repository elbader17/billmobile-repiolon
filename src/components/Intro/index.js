import React from 'react';
import { View, Text, Image } from 'react-native';
import AppIntroSlider from 'react-native-app-intro-slider';
import {SLIDES_INTRO} from '../../constants/slides_intro';
import style from './style';

class Intro extends React.Component {
  _renderItem = (item) => {
    return (
      <View style={style.container} >
        <View style={style.silde}>
          <View style={style.box1}>
            <Image source={ item.image } style={ style.imageHeader } />  
          </View>
          <View style={style.box2}>
            <Text style={style.textTittle}>{item.text}</Text>
            <Text style={style.textDescription}>{item.description}</Text>
          </View>
        </View>
      </View>
    );
  }
  _onDone = () => {
    this.props.navigation.navigate('Login')
  }
  render() {
    return ( 
      <AppIntroSlider 
        renderItem={this._renderItem} 
        slides={SLIDES_INTRO} 
        onDone={this._onDone} 
        bottomButton
        buttonStyle={style.button}
        buttonTextStyle={style.buttonText}
        doneLabel='Listo'
        nextLabel='Siguiente'
        activeDotStyle={style.activeDotStyle}
        dotStyle={style.dotStyle}
      />
    )
  }
}

export default Intro;