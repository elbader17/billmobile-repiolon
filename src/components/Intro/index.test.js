import React from 'react';
import Intro from './index';
import AppIntroSlider from 'react-native-app-intro-slider';
import renderer from 'react-test-renderer';

test('render correctly component Intro', () => {
  const component = renderer.create(<Intro />);  
  expect(component.toJSON()).toMatchSnapshot();
});

test('onDone function is being called once', () => {
  const fn = jest.fn();
  const component = renderer.create(
    <AppIntroSlider 
      renderItem={()=>{}} 
      slides={{}} 
      onDone={fn} 
      bottomButton
      buttonStyle={{}}
      buttonTextStyle={{}}
      doneLabel=''
      nextLabel=''
      activeDotStyle={{}}
      dotStyle={{}}
    />
  );
  const instance = component.getInstance();
  instance.props.onDone();
  expect(fn.mock.calls.length).toBe(1);
});
