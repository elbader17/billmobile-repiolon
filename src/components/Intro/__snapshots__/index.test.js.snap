// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly component Intro 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bottomButton={true}
    bounces={false}
    data={
      Array [
        Object {
          "backgroundColor": "#59b2ab",
          "description": "Como cuando lo era cuando lo hacias en papel, ágil para vos y para tus clientes.",
          "text": "Facturar sin demoras",
          "title": "1",
        },
        Object {
          "backgroundColor": "#febe29",
          "description": "No tendrás que cargar la misma información dos veces. Nuestro sistema guarda y ayuda a hacer más ágil tu facturación.",
          "text": "Hacemos práctica la facturación",
          "title": "2",
        },
        Object {
          "backgroundColor": "#22bcb5",
          "description": "Nuestra política de privacidad es simple: No compartimos ninguna información personal o impositiva tuya.",
          "text": "Informacón segura",
          "title": "3",
        },
      ]
    }
    disableVirtualization={false}
    extraData={750}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    paginationStyle={null}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showDoneButton={true}
    showNextButton={true}
    showsHorizontalScrollIndicator={false}
    slides={
      Array [
        Object {
          "backgroundColor": "#59b2ab",
          "description": "Como cuando lo era cuando lo hacias en papel, ágil para vos y para tus clientes.",
          "text": "Facturar sin demoras",
          "title": "1",
        },
        Object {
          "backgroundColor": "#febe29",
          "description": "No tendrás que cargar la misma información dos veces. Nuestro sistema guarda y ayuda a hacer más ágil tu facturación.",
          "text": "Hacemos práctica la facturación",
          "title": "2",
        },
        Object {
          "backgroundColor": "#22bcb5",
          "description": "Nuestra política de privacidad es simple: No compartimos ninguna información personal o impositiva tuya.",
          "text": "Informacón segura",
          "title": "3",
        },
      ]
    }
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3687d1",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0.8,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": undefined,
                    "fontSize": 78,
                  }
                }
              >
                1
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.2,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                Facturar sin demoras
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingHorizontal": 50,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                    "lineHeight": 21,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                Como cuando lo era cuando lo hacias en papel, ágil para vos y para tus clientes.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3687d1",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0.8,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": undefined,
                    "fontSize": 78,
                  }
                }
              >
                2
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.2,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                Hacemos práctica la facturación
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingHorizontal": 50,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                    "lineHeight": 21,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                No tendrás que cargar la misma información dos veces. Nuestro sistema guarda y ayuda a hacer más ágil tu facturación.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3687d1",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0.8,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": undefined,
                    "fontSize": 78,
                  }
                }
              >
                3
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.2,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                Informacón segura
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingHorizontal": 50,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                    "lineHeight": 21,
                    "textAlign": "center",
                    "width": 600,
                  }
                }
              >
                Nuestra política de privacidad es simple: No compartimos ninguna información personal o impositiva tuya.
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 16,
          "left": 16,
          "position": "absolute",
          "right": 16,
        },
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 16,
          "justifyContent": "center",
          "margin": 16,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "height": 9,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 9,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "borderRadius": 5,
            "height": 9,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 9,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "borderRadius": 5,
            "height": 9,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 9,
          }
        }
      />
    </View>
    <View
      style={false}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, .3)",
                "flex": 1,
                "justifyContent": "center",
              },
              false,
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#fff",
                "borderRadius": 2,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 12,
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "Lato-Semibold",
                  "fontSize": 15,
                },
              ]
            }
          >
            Siguiente
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
